test/MicrosoftTeams.spec.ts

```diff
@@ -1118,70 +1118,77 @@ describe("MicrosoftTeams", () => {
-        authenticationParams.url.toLowerCase() + "/"
-      );
-      expect(message.args[1]).toBe(authenticationParams.width);
-      expect(message.args[2]).toBe(authenticationParams.height);
+    ["android", "ios", "desktop"].forEach(hostClientType => {
+      it(`should successfully pop up the auth window in the ${hostClientType} client`, () => {
+        initializeWithContext("content", hostClientType);
+
+        let authenticationParams = {
+          url: "https://someUrl",
+          width: 100,
+          height: 200
+        };
+        microsoftTeams.authentication.authenticate(authenticationParams);
+
+        let message = findMessageByFunc("authentication.authenticate");
+        expect(message).not.toBeNull();
+        expect(message.args.length).toBe(3);
+        expect(message.args[0]).toBe(
+          authenticationParams.url.toLowerCase() + "/"
+        );
+        expect(message.args[1]).toBe(authenticationParams.width);
+        expect(message.args[2]).toBe(authenticationParams.height);
+      });
     });
 
-    it("should successfully handle auth success in the desktop client", () => {
-      initializeWithContext("content", "desktop");
+    ["android", "ios", "desktop"].forEach(hostClientType => {
+      it(`should successfully handle auth success in the ${hostClientType} client`, () => {
+        initializeWithContext("content", hostClientType);
 
-      let successResult: string;
-      let failureReason: string;
-      let authenticationParams = {
-        url: "https://someUrl",
-        width: 100,
-        height: 200,
-        successCallback: (result: string) => (successResult = result),
-        failureCallback: (reason: string) => (failureReason = reason)
-      };
-      microsoftTeams.authentication.authenticate(authenticationParams);
+        let successResult: string;
+        let failureReason: string;
+        let authenticationParams = {
+          url: "https://someUrl",
+          width: 100,
+          height: 200,
+          successCallback: (result: string) => (successResult = result),
+          failureCallback: (reason: string) => (failureReason = reason)
+        };
+        microsoftTeams.authentication.authenticate(authenticationParams);
 
-      let message = findMessageByFunc("authentication.authenticate");
-      expect(message).not.toBeNull();
+        let message = findMessageByFunc("authentication.authenticate");
+        expect(message).not.toBeNull();
 
-      respondToMessage(message, true, "someResult");
+        respondToMessage(message, true, "someResult");
 
-      expect(successResult).toBe("someResult");
-      expect(failureReason).toBeUndefined();
+        expect(successResult).toBe("someResult");
+        expect(failureReason).toBeUndefined();
+      });
     });
 
-    it("should successfully handle auth failure in the desktop client", () => {
-      initializeWithContext("content", "desktop");
+    ["android", "ios", "desktop"].forEach(hostClientType => {
```

Comment:
NIT: You don’t need to repeat the loops; just move all of these test cases into the first loop above.

Reply:
true... was just thinking that